src/edu/csus/ecs/pc2/list/ListUtilities.java

```diff
@@ -0,0 +1,224 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.list;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import edu.csus.ecs.pc2.core.LanguageUtilities;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.model.Language;
+import edu.csus.ecs.pc2.core.model.Problem;
+import edu.csus.ecs.pc2.imports.ccs.IContestLoader;
+import edu.csus.ecs.pc2.validator.clicsValidator.ClicsValidator;
+
+/**
+ * A set of utilities that operate on lists.
+ *
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class ListUtilities {
+
+	/**
+     * Returns list of files that match the SubmissionSolutionList (judging types location/list)
+     *
+     * @param files list of files to be filtered
+     * @param submissionSolutionList List of types to accept
+     * @return
+     */
+    public static List<File> filterByJudgingTypes(List<File> files, List<String> submissionSolutionList) {
+
+        List<File> newFileList = new ArrayList<File>();
+        int idx;
+
+        for (File file : files) {
+            String filePath = file.getAbsolutePath();
+            for(String subSol : submissionSolutionList) {
+                idx = subSol.indexOf(':');
+                if(idx > 0) {
+                    subSol = subSol.substring(0, idx).trim();
+                }
+                if (filePath.contains(File.separator + subSol + File.separator)) {
+                    newFileList.add(file);
+                }
+            }
+        }
+
+        return newFileList;
+    }
+
+    /**
+     * Returns list of all files under dir path.
+     *
+     * @param dirName location to fetch files from
+     * @return list of files in and under dirName
+     */
+	// TODO REFACTOR replace findAll from QuickSubmitter with this method.
+    public static List<File> findAllFiles(String dirName) {
+
+        List<File> files = new ArrayList<>();
+
+        File dir = new File(dirName);
+        File[] entries = dir.listFiles();
+
+        if (entries == null) {
+            return files;
+        }
+
+        for (File f : entries) {
+            if (f.isDirectory()) {
+                List<File> subList = findAllFiles(f.getAbsolutePath());
+                files.addAll(subList);
+            } else {
+                files.add(f);
+            }
+        }
+
+        return files;
+    }
+
+	/**
+	 * Get all CDP judge's sample submission filenames.
+	 *
+	 * Example files under config\sumit\submissions
+	 *
+	 * @param mycontest
+	 * @param directoryName CDP config/ dir or CDP base directory
+	 * @return list of judges sample submissions
+	 */
+	// TODO REFACTOR remove getAllCDPsubmissionFileNames from QuickSubmitter
+	public static List<File> getAllJudgeSampleSubmissionFilenamesFromCDP(IInternalContest mycontest, String directoryName) {
+
```

Comment:
this is indented wrong

Reply:
Fixed and pushed. There were residual tabs from the original code.